src/pages/ClaimVestedTokens/ClaimVestedTokens.tsx

```diff
@@ -99,12 +108,30 @@ const ClaimVestedTokens: React.FC = () => {
 		vesting.releasableTokens,
 		vesting.awardedTokens,
 		vesting.vestingParams,
+		contractNumber,
 	]);
 
+	//If full fetch happens, change to action button
+	useEffect(() => {
+		setLoading(false);
+	}, [vesting.fullFetch]);
+
+	//If half fetch happens and hasnt claimed, change to action button
+	useEffect(() => {
+		if (vesting.hasClaimed === false) setLoading(false);
+	}, [vesting.partialFetch]);
+
 	const addWildToMetamask = () => suggestWildToken(context.library);
 
 	const toNumber = (amount: any) => Number(ethers.utils.formatEther(amount));
 
+	//toggle contract, and set the loading icon on
+	const contractToggle = () => {
+		setLoading(true);
+		setContractNumber(contractNumber + 1);
```

Comment:
Minor fix here; instead of calling `setContractNumber` twice (in the case of loop) pre-calculate what it should be set to, then only set it once.